### PR TITLE
Added plugin to read Windows performance counters

### PR DESCRIPTION
**Pull request by TheFlyingCorpse**
Plugin that reads performance counters from Windows, a lot of examples included to get an idea how to use this with IIS and AD to start with.
The plugin is flexible so you can have a master configuration to push out to all, it will only submit from each host what is matching to your configuration.

### CODE REVIEW

File: plugins/inputs/win_perfcounters/win_perfcounters.go [hunk 1]
```diff
@@ -0,0 +1,375 @@
+// +build windows
+
+package win_perfcounters
+
+import (
+	"fmt"
+	"strings"
+	"syscall"
+	//"time"
```
Comment (sparrc) [inline plugins/inputs/win_perfcounters/win_perfcounters.go]: remove commented import

File: plugins/inputs/win_perfcounters/win_perfcounters.go [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+	"unsafe"
+
+	"os"
+	"os/signal"
+
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/lxn/win"
+)
+
+var sampleConfig string = `#  By default this plugin gathers IO
```
Comment (sparrc) [inline plugins/inputs/win_perfcounters/win_perfcounters.go]: Why does the plugin gather IO by default?

File: plugins/inputs/win_perfcounters/win_perfcounters.go [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+#    Measurement = "win_ad"
+#    #IncludeTotal=false #Set to true to include _Total instance when querying for all (*).
+#  [[inputs.win_perfcounters.object]]
+#    # AD
+#    ObjectName = "Database"
+#    Instances = ["*"]
+#    Counters = ["Database Cache % Hit","Database Cache Page Fault Stalls/sec","Database Cache Page Faults/sec","Database Cache Size"]
+#    Measurement = "win_db"
+#    #IncludeTotal=false #Set to true to include _Total instance when querying for all (*).
+#
```
Comment (sparrc) [inline plugins/inputs/win_perfcounters/win_perfcounters.go]: remove extra lines at the end here

**Comment by sparrc** [comment]
Thanks a bunch @TheFlyingCorpse!

A couple things to address:

1. Write a README please!
1. Can you unit test some or all of this?
1. Put a dummy file called `win_perfcounters_notwindows.go` with the `// +build !windows` flag in the directory, otherwise unix builds will complain that there are no buildable source files in the directory.
1. Are all of those examples necessary for the config file? Could you maybe cut those down by at least half and then put the rest in a README file?
1. Try to reduce line-length to 80 characters throughout
1. If you happen to have any useful "recipes" of combinations of perf counters that you commonly use, that would be great to have in the README as well. Such as "cpu monitoring" and then give an example of the objects you might want to monitor the cpu.

**Comment by TheFlyingCorpse** [comment]
I'm still working on the testcases.

**Comment by sparrc** [comment]
@TheFlyingCorpse Is this ready to merge?

**Comment by TheFlyingCorpse** [comment]
@sparrc - Yes, I have formatted the code to the best of my ability and added unit tests for both the configuration parsing and doing test queries on Windows.

**Comment by sparrc** [comment]
@TheFlyingCorpse can you please squash your commits and rebase off master?

**Comment by TheFlyingCorpse** [comment]
@sparrc, done!

**Comment by sparrc** [comment]
thanks @TheFlyingCorpse! this is in master now

**Comment by sparrc** [comment]
note: I changed the name to `win_perf_counters` for consistency

**Pull request closed without merging**